### ISSUE TITLE
json-schema driven app fails to compile with typescript's `"noImplicitAny": true` option

### ISSUE THREAD

**Issue by Liooo** [ngx-formly/ngx-formly#2218]
**Description**

When adding `"noImplicitAny": true` in tsconfig.json, the compilation fails with the following error.

```
ERROR in node_modules/@ngx-formly/core/json-schema/formly-json-schema.service.d.ts(2,29): error TS7016: Could not find a declaration file for module 'json-schema'. '/path/to/node_modules/json-schema/lib/validate.js' implicitly has an 'any' type.
  Try `npm install @types/json-schema` if it exists or add a new declaration (.d.ts) file containing `declare module 'json-schema';`
```

The compile succeeds and app works without a problem, when noImplicitAny is not specified.

**Minimal Reproduction**

https://github.com/Liooo/formly-jsonschema

**Your Environment**

in https://github.com/Liooo/formly-jsonschema/blob/master/package.json
```
    ...
    "@angular/animations": "~7.2.0",
    "@angular/cdk": "~7.3.7",
    "@angular/common": "~7.2.0",
    "@angular/compiler": "~7.2.0",
    "@angular/core": "~7.2.0",
    "@angular/forms": "~7.2.0",
    "@angular/material": "^7.3.7",
    "@angular/platform-browser": "~7.2.0",
    "@angular/platform-browser-dynamic": "~7.2.0",
    "@angular/router": "~7.2.0",
    "@ngx-formly/core": "^5.0.0",
    "@ngx-formly/material": "^5.0.0",
    "@ngx-formly/schematics": "^5.6.1",
    ...
```

**Additional context**

Basically just did something like the following:

```bash
npm install -g @angular/cli@7.3.9

ng new my-app
cd my-app/

ng add  @angular/material
ng add @ngx-formly/schematics --ui-theme=bootstrap

# then, add minimum code to reproduce `simple` schema from https://formly.dev/examples/advanced/json-schema
```

**Comment by aitboudad** [comment]
have you tried to install `json-schame` declaration:
```ts
npm install @types/json-schema --save-dev
```

**Comment by Liooo** [comment]
@aitboudad 

Thanks for the quick response, actually I did it last night and could get the repo to work.

Wasn't quite sure about the handling of 3rd party types inside the libraries in general, and found something like this: https://github.com/Microsoft/TypeScript/issues/18974#issuecomment-334802090 , so probably this better be declared in ngx-formly's dependency?

Also react-jsonschema-form is doing the same:
https://github.com/rjsf-team/react-jsonschema-form/blob/0b638420c9834993788e1f313677dc6992360b50/packages/core/package.json#L46

If this sounds good, I'll send a PR adding @types/json-schema in @ngx-formly/core's dependency, what do you think?

**Comment by aitboudad** [comment]
`json-schema` is an optional dependency in `core` so I'd prefer to not add it.

My plan is to move `json-schema` into a separate sub-package `@ngx-formly/json-schema` so if that happends `@types/json-schema` will be added as deps.

Thank you anyway :+1:

**Comment by Liooo** [comment]
@aitboudad sounds good 👍
